concordium-consensus/src/Concordium/GRPC2.hs

```diff
@@ -1960,6 +2000,119 @@ getTokenomicsInfoV2 cptr blockType blockHashPtr outHash outVec copierCbk = do
     result <- runMVR (Q.getRewardStatus bhi) mvr
     returnMessageWithBlock (copier outVec) outHash result
 
+getPoolDelegatorsV2 ::
+    StablePtr Ext.ConsensusRunner ->
+    Ptr SenderChannel ->
+    -- |Block type.
+    Word8 ->
+    -- |Block hash.
+    Ptr Word8 ->
+    -- |Baker id.
+    Word64 ->
+    -- |Out pointer for writing the block hash that was used.
+    Ptr Word8 ->
+    FunPtr ChannelSendCallback ->
+    IO Int64
+getPoolDelegatorsV2 cptr channel blockType blockHashPtr bakerId outHash cbk = do
+    Ext.ConsensusRunner mvr <- deRefStablePtr cptr
+    let sender = callChannelSendCallback cbk
+    bhi <- decodeBlockHashInput blockType blockHashPtr
+    (bh, mDelegators) <- runMVR (Q.getPoolDelegators bhi (Just $ fromIntegral bakerId)) mvr
+    case mDelegators of
+        Nothing -> return $ queryResultCode QRInvalidArgument
+        Just Nothing -> return $ queryResultCode QRNotFound
+        Just (Just delegators) -> do
+            copyHashTo outHash bh
+            _ <- enqueueMessages (sender channel) delegators
+            return (queryResultCode QRSuccess)
```

Comment:
As you mentioned yourself, perhaps we should have an `Either` with a custom error here instead of `Maybe (Maybe a))`.